avocado/plugins/xunit.py

```diff
@@ -90,7 +90,7 @@ def _create_failure_or_error(self, document, test, element_type,
     def _render(self, result, max_test_log_size):
         document = Document()
         testsuite = document.createElement('testsuite')
-        testsuite.setAttribute('name', 'avocado')
+        testsuite.setAttribute('name', os.path.basename(os.path.dirname(result.logfile)))
```

Comment:
Where should I see this in Jenkins' display of the xunit result?

Reply:
Ahh.  Yes.  I see.  I have to click the test result link from the _testReport/_ url.  I typically save the forward (clicking the test links) and back action by just clicking the those big plus (and minus) sign expander/collapsers.  Apart from that new _(from jobs YYYY-MM-DDTHH.MM-UUID)_ I didn't see anything that the cost of clicking links (and having to go back) was worth.

Reply:
Anyway looking at junit reference: https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd there is no other way unless we want to inject it into stdout/stderr/name/classname/message/traceback. As Jenkins is our primary target for xunit and it only provides the testsuite-name buried deeply in results, we might consider extending these. Ideal would be the `classname`, but I'm not sure about the consequences (it'd better differentiate the test results for multiple jobs while it'd avoid grouping the results in case we want group multiple-job-results together).

What I mean:

avocado run suite.yaml --some-setting
avocado run suite.yaml --other-setting

would show individual results for each avocado execution while now it groups the same tests from both executions into a single classname.

Actually when I started I wanted to use classname as jobname, because it was more natural, but when I looked at other tools they like grouping results per classname. It's a tough call...

Reply:
... I mean in java testing I understand you might want to get aggregated results per failed class, but in generic testing framework like Avocado it usually doesn't make sense (and often I need to look at ExternalRunner, FailTest, ... results individually only to see regressions/improvements). When adding xunit results we simply followed the description in junit, which is classname but it could be changed.

Reply:
Yeah.  That is nice.  I can appreciate the limitations that junit puts on being able to do any of this in a nice way.